pwa-devdocs/src/technologies/basic-concepts/internationalization/index.md

```diff
@@ -0,0 +1,157 @@
+
+Both approach have their pros and cons, and developers are free to choose which approach works for them when they develop their own components and storefronts.
+
+### Message syntax
+
+The i18n framework accepts the same [message syntax][] as the underlying `react-intl` library.
+Along with static text, this syntax supports variables, dates, and even conditional formatting.
+
+To translate text with variables, pass in a mapping object to the `values` prop in the `FormattedMessage` component.
+
+{% raw %}
+
+```jsx
+return(
+    <FormattedMessage
+        id="component.greeting"
+        defaultMessage="Hi, {name}"
+        values={{
+            name: 'Veronica',
+        }}
+    />
+)
+```
+
+{% endraw %}
+
+When using the `formatMessage()` function, pass in the mapping object as the second parameter.
+
+```jsx
+const text = formatMessage(
+    { id:"component.greeting", defaultMessage="Hi, {name}"},
+    { name: currentUser.firstname }
+);
+```
+
+For more details, see the [message syntax][] documentation at FormatJS.
+
+## Language packages and plugins
+
+Language packages provide translation data for one or more locales.
+They are also used to override the text in the same locale.
+
+Unlike the core Magento application, which install language packages through Composer, PWA Studio storefronts install language packages as NPM dependencies.
+
+An NPM dependency is a language package if it meets the following criteria:
+
+- The package contains an intercept file
+- The intercept file sets the special feature `i18n` flag to `true` for the package
+- The package contains an `i18n` directory
+- The `i18n` directory contains a dictionary file with a locale formatted name
+
+<!-- TODO: Create an in-depth tutorial for creating a language package extension -->
+
+## Build process
+
+To optimize runtime performance, the i18n framework compiles all the translation data during the build process.
+
+The following is a high level summary of the actions the i18n framework takes to compile the translation data:
+
+1. Scans the `i18n` directory in the project and installed language packages for `.json` files that have a locale formatted file name
+2. Generates an object with locales as keys that map to an array of files matching that locale
+3. Merges the files in the arrays to create a single dictionary object for a locale
+4. Creates a virtual module from this object that exposes a `__fetchLocaleData__` function
+5. Generates a dynamic import in the application for the virtual module
+
+## Runtime process
+
+During runtime, the `LocaleProvider` component uses the `__fetchLocaleData__` function to get the correct translation data for the current locale.
+
+If a components changes the value of the current locale during runtime, the framework sends a GraphQL query to verify the new value.
+Even if you install a language package plugin for a locale, you must enable the locale on the backend to use the translations in the storefront UI.
```

Comment:
What is this concerning? We don't do any GraphQL queries for individual string changes during development. Be good if you can clarify which feature you mean by this.

Reply:
I wanted to make sure we covered our bases in terms of what minimum requirements are met in order to enable and use translations in the UI. If a user makes an `i18n/fr_FR.json` they can't just display French right? They'd have to enable it in the backend so that it shows up in the locale switcher that @eug123 has created.

Reply:
additional clarification provided

Reply:
Thanks for explaining @sirugh!